Graphics/GraphicsEngineD3D11/src/ShaderD3D11Impl.cpp

```diff
@@ -34,24 +34,24 @@
 namespace Diligent
 {
 
-static const std::string HLSLVersionToShaderModelString(const ShaderVersion& Version, Uint8 MaxMajorRevision, Uint8 MaxMinorRevision)
+static const Uint8 HLSLVersionToShaderModelString(const ShaderVersion& Version, Uint8 MaxMajorRevision, Uint8 MaxMinorRevision)
 {
-    std::string ModelStr;
+    Uint8 ModelVer;
     if (Version.Major > MaxMajorRevision || Version.Major == MaxMajorRevision && Version.Minor > MaxMinorRevision)
     {
-        ModelStr = std::to_string(Uint32{MaxMajorRevision}) + '_' + std::to_string(Uint32{MaxMinorRevision});
+        ModelVer = Uint8((MaxMajorRevision << 4) | MaxMinorRevision);
```

Comment:
Brace initialization Uint8{} is preferable, because Uint8() is in fact a c-style cast, which is very unsafe: Uint8(nullptr) will compile just fine

Reply:
c-style cast is (Uint8)().
Uint8() and Uint8{} is converting constructor.
You can add GCC warning `-Wold-style-cast` to warn about c-style casts.

In that place I will use ShaderVersion instead of Uint8.

Reply:
Yes you are actually correct. Still, brace initialization is preferable

Reply:
On MSVC this does make a difference:
https://godbolt.org/z/qK44zj
https://godbolt.org/z/Kjx3ET